torchgeo/datasets/mapinwild.py

```diff
@@ -0,0 +1,513 @@
+        "s2_summer_part2.zip": "5b5816bbd32987619bf72cde5cacd032",
+        "s2_winter_part1.zip": "ca958f7cd98e37cb59d6f3877573ee6d",
+        "s2_winter_part2.zip": "e7aacb0806d6d619b6abc408e6b09fdc",
+    }
+
+    mask_palette = {1: (0, 153, 0), 0: (255, 255, 255)}
+
+    wc_palette = {
+        10: (0, 160, 0),
+        20: (150, 100, 0),
+        30: (255, 180, 0),
+        40: (255, 255, 100),
+        50: (195, 20, 0),
+        60: (255, 245, 215),
+        70: (255, 255, 255),
+        80: (0, 70, 200),
+        90: (0, 220, 130),
+        95: (0, 150, 120),
+        100: (255, 235, 175),
+    }
+
+    def __init__(
+        self,
+        root: str = "data",
+        modality: list[str] = ["mask", "esa_wc", "viirs"],
+        split: str = "train",
+        transforms: Optional[Callable[[dict[str, Tensor]], dict[str, Tensor]]] = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new MapInWild dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            modality: the modality to download. Choose from: "mask", "esa_wc",
+                "viirs", "s1", "s2_temporal_subset", "s2_[enter_season]".
+            split: one of "train", "val", or "test"
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            AssertionError: if ``split`` argument is invalid
+            ImportError: if pandas is not installed
+        """
+        assert split in ["train", "validation", "test"]
+
+        self.band_indices = torch.tensor(
+            [self.BAND_SETS["s2"].index(b) for b in self.BAND_SETS["s2"]]
+        ).long()
+
+        self.checksum = checksum
+        self.root = root
+        self.transforms = transforms
+        self.modality = modality
+        self.download = download
+        self._verify_split()
+
+        try:
+            import pandas as pd  # noqa: F401
+        except ImportError:
+            raise ImportError(
+                "pandas is not installed and is required to use this dataset"
+            )
+
+        split_dataframe = pd.read_csv(os.path.join(self.root, "split_IDs.csv"))
+        self.ids = split_dataframe[split].dropna().values.tolist()
+        self.ids = [int(i) for i in self.ids]
+
+        for modal in modality:
+            for modality_link in self.modality_urls[modal]:
+                self._verify(modality_link, self.md5s[os.path.split(modality_link)[1]])
+                if checksum:
+                    if not self._check_integrity(
+                        modality_link, self.md5s[os.path.split(modality_link)[1]]
+                    ):
+                        raise RuntimeError("Dataset not found or corrupted.")
```

Comment:
We usually handle checksumming in `_verify` so that `__init__` isn't super long and confusing